packages/framework/client-logger/external-telemetry/src/container/containerTelemetry.ts

```diff
@@ -0,0 +1,170 @@
+	 */
+	clientId?: string;
+	/**
+	 * The id of the container this resolved url is for.
+	 * Will be undefined only when the container is not attached.
+	 */
+	documentId?: string;
+}
+
+/**
+ * The container "connected" telemetry event.
+ * It is produced from an internal Fluid container system event {@link @fluidframework/container-definitions#IContainerEvents} which is emitted when the {@link @fluidframework/container-definitions#IContainer} completes connecting to the Fluid service.
+ *
+ * @remarks Reflects connection state changes against the (delta) service acknowledging ops/edits.
+ *
+ * @see
+ *
+ * - {@link @fluidframework/container-definitions#IContainer.connectionState}
+ *
+ * - {@link @fluidframework/container-definitions#IContainer.connect}
+ * @see {@link @fluidframework/container-definitions#IContainer.connectionState}
+ *
+ * @beta
+ */
+export interface ContainerConnectedTelemetry extends IContainerTelemetry {
+	eventName: "container.connected";
+}
+
+/**
+ * The container "disconnected" telemetry event. This telemetry is produced from an internal Fluid container system event
+ * {@link @fluidframework/container-definitions#IContainerEvents} which is emitted when the {@link @fluidframework/container-definitions#IContainer}
+ * becomes disconnected from the Fluid service.
+ *
+ * @remarks Reflects connection state changes against the (delta) service acknowledging ops/edits.
+ *
+ * @see
+ *
+ * - {@link @fluidframework/container-definitions#IContainer.connectionState}
+ *
+ * - {@link @fluidframework/container-definitions#IContainer.disconnect}
+ *
+ * @beta
+ */
+export interface ContainerDisconnectedTelemetry extends IContainerTelemetry {
+	eventName: "container.disconnected";
+}
+
+/**
+ * The container "closed" telemetry event. This telemetry is produced from an internal Fluid container system event
+ * {@link @fluidframework/container-definitions#IContainerEvents} which is emitted when the {@link @fluidframework/container-definitions#IContainer}
+ * is closed, which permanently disables the container.
+ *
+ * @remarks Listener parameters:
+ *
+ * - `error`: If the container was closed due to error, this will contain details about the error that caused it.
+ *
+ * @see {@link @fluidframework/container-definitions#IContainer.close}
+ *
+ * @beta
+ */
+export interface ContainerClosedTelemetry extends IContainerTelemetry {
+	eventName: "container.closed";
+	error?: ICriticalContainerError;
+}
+
+/**
+ * The container "attaching" telemetry event. This telemetry is produced from an internal Fluid container system event
+ * {@link @fluidframework/container-definitions#IContainerEvents} which is emitted when the {@link @fluidframework/container-definitions#IContainer}'s
+ * {@link @fluidframework/container-definitions#AttachState.Attaching | attaching} process is complete and the container is {@link @fluidframework/container-definitions#AttachState.Attached | attached} to the Fluid service.
```

Comment:
I think there's a bit of mixup between in the docs for attaching and attached.

Reply:
Fixed